src/modules/test/ManagementModule.cpp

```diff
@@ -0,0 +1,516 @@
+        status = EINVAL;
+    }
+
+    // Optional fields
+
+    // Version Patch
+    if (object.HasMember(g_mmiGetInfoVersionPatch))
+    {
+        if (object[g_mmiGetInfoVersionPatch].IsInt())
+        {
+            info.version.patch = object[g_mmiGetInfoVersionPatch].GetInt();
+        }
+        else
+        {
+            OsConfigLogError(NULL, "Module info field '%s' is not an integer", g_mmiGetInfoVersionPatch);
+        }
+    }
+
+    // Version Tweak
+    if (object.HasMember(g_mmiGetInfoVersionTweak))
+    {
+        if (object[g_mmiGetInfoVersionTweak].IsInt())
+        {
+            info.version.tweak = object[g_mmiGetInfoVersionTweak].GetInt();
+        }
+        else
+        {
+            OsConfigLogError(NULL, "Module info field '%s' is not an integer", g_mmiGetInfoVersionTweak);
+        }
+    }
+
+
+    // License URI
+    if (object.HasMember(g_mmiGetInfoLicenseUri))
+    {
+        if (object[g_mmiGetInfoLicenseUri].IsString())
+        {
+            info.licenseUri = object[g_mmiGetInfoLicenseUri].GetString();
+        }
+        else
+        {
+            OsConfigLogError(NULL, "Module info field '%s' is not a string", g_mmiGetInfoLicenseUri);
+        }
+    }
+
+    // Project URI
+    if (object.HasMember(g_mmiGetInfoProjectUri))
+    {
+        if (object[g_mmiGetInfoProjectUri].IsString())
+        {
+            info.projectUri = object[g_mmiGetInfoProjectUri].GetString();
+        }
+        else
+        {
+            OsConfigLogError(NULL, "Module info field '%s' is not a string", g_mmiGetInfoProjectUri);
+        }
+    }
+
+    // User Account
+    if (object.HasMember(g_mmiGetInfoUserAccount))
+    {
+        if (object[g_mmiGetInfoUserAccount].IsUint())
+        {
+            info.userAccount = object[g_mmiGetInfoUserAccount].GetUint();
+        }
+        else
+        {
+            OsConfigLogError(NULL, "Module info field '%s' is not an unsigned integer", g_mmiGetInfoUserAccount);
+        }
+    }
+
+    return status;
+}
+
+MmiSession::MmiSession(std::shared_ptr<ManagementModule> module, const std::string& clientName, unsigned int maxPayloadSizeBytes) :
+    m_clientName(clientName),
+    m_maxPayloadSizeBytes(maxPayloadSizeBytes),
+    m_module(module),
+    m_mmiHandle(nullptr) {}
+
+MmiSession::~MmiSession()
+{
+    Close();
+}
+
+int MmiSession::Open()
+{
+    int status = 0;
+
+    if (nullptr != m_module)
+    {
+        if (nullptr == m_mmiHandle)
+        {
+            if (nullptr == (m_mmiHandle = m_module->CallMmiOpen(m_clientName.c_str(), m_maxPayloadSizeBytes)))
+            {
+                OsConfigLogError(NULL, "Failed to open MMI session for client '%s'", m_clientName.c_str());
+            }
+        }
+        else
+        {
+            status = EINVAL;
+            if (IsFullLoggingEnabled())
+            {
+                OsConfigLogError(NULL, "MMI session already open");
+            }
```

Comment:
Change all OSConfig traces in this code to test traces.

Don't link to logging library unless we will drop support for GTest and replace with that.